pkg/providers/aws/provider_redis.go

```diff
@@ -271,11 +265,24 @@ func (p *RedisProvider) createElasticacheCluster(ctx context.Context, r *v1alpha
 	var replicationGroupClusters []elasticache.CacheCluster
 	for _, checkedCluster := range cacheClustersOutput.CacheClusters {
 		cluster := *checkedCluster
-		if resources.SafeStringDereference(cluster.ReplicationGroupId) == *foundCache.ReplicationGroupId {
-			replicationGroupClusters = append(replicationGroupClusters, *checkedCluster)
+		if checkedCluster.ReplicationGroupId != nil && foundCache.ReplicationGroupId != nil {
```

Comment:
The safeStringDereference below handles this being nil.
The function returns "" if it's nil

Reply:
Yeah but it doesn't handle the case if `foundCache` is nil. 

I have no strong opinion on removing the nil check here or doing the stringDereference, or leaving both.